scripts/utils/trading_strategy_helpers.js

```diff
@@ -372,24 +372,20 @@ withdrawal of or to withdraw the desired funds
    */
   const buildTransferApproveDepositFromList = async function(masterAddress, depositList, debug = false) {
     const log = debug ? (...a) => console.log(...a) : () => {}
-    const ERC20 = artifacts.require("ERC20Detailed")
 
     let transactions = []
     // TODO - make cumulative sum of deposits by token and assert that masterSafe has enough for the tranfer
-    // TODO - make deposit list easier so that we dont' have to query the token every time.
     for (const deposit of depositList) {
       assert(
         await isOnlySafeOwner(masterAddress, deposit.bracketAddress),
         "All depositors must be owned only by the master Safe"
       )
-      const depositToken = await ERC20.at(deposit.tokenAddress)
-      const tokenSymbol = await depositToken.symbol.call()
-      const tokenDecimals = await depositToken.decimals.call()
-      const unitAmount = fromErc20Units(deposit.amount, tokenDecimals)
+      const tokenInfo = await fetchTokenInfoAtAddresses([deposit.tokenAddress], debug)
```

Comment:
The syntax is a bit different: `fetchTokenInfoAtAddresses` takes an array of addresses and returns a dictionary of promises, indexed by the address of the token. So this should be something like:
```
const tokenInfoPromises = fetchTokenInfoAtAddresses([deposit.tokenAddress], debug)
const tokenInfo = await tokenInfoPromises[deposit.tokenAddress]
```

Reply:
I'm very surprised that the checks have passed. How is this possible?

Reply:
Your syntax would be the right one for the function `fetchTokenInfoFromExchange`
But I am using `fetchTokenInfoAtAddresses`

Reply:
Mmmh, they look consistent to me, the output `tokenPromises` is defined as a dictionary and every token is identified by its address.
Maybe the checks passed because here we use token info just for logging, which is not part of any test?

Reply:
Ups you are right.
It just works, because the dictionary has only one object and so it is selected automatically, I guess. Let me change it to have it more explicit.